src/logging.cc

```diff
@@ -1314,12 +1322,24 @@ void LogCleaner::Disable() {
   enabled_ = false;
 }
 
+void LogCleaner::UpdateCleanUpTime() {
+  const int64 next = (FLAGS_logcleansecs
+                      * static_cast<int64>(1000000));  // in usec
+  next_cleanup_time_ = CycleClock_Now() + UsecToCycles(next);
+}
+
 void LogCleaner::Run(bool base_filename_selected,
                      const string& base_filename,
-                     const string& filename_extension) const {
+                     const string& filename_extension) {
   assert(enabled_);
   assert(!base_filename_selected || !base_filename.empty());
 
+  // avoid scanning logs too frequently
+  if (CycleClock_Now() < next_cleanup_time_) {
+    return;
```

Comment:
This line is not covered by any tests. Could you please either extend an existing test to cover the line or remove it altogether?

Reply:
Thanks for the reminder, I've extended an existing test in file : cleanup_with_absolute_prefix_unittest.cc